blockstack_client/keys.py

```diff
@@ -377,107 +357,102 @@ def get_data_or_owner_privkey( user_zonefile, owner_address, wallet_keys=None, c
     """
 
     # generate the mutable zonefile
-    data_privkey = get_data_privkey( user_zonefile, wallet_keys=wallet_keys, config_path=config_path )
-    if data_privkey is None:
-        # This is legacy code here.  The only time this should happen is
-        # when the user has a single owner key, and does not have a 
-        # separate data key.
-        log.warn("No data private key set.  Falling back to owner keypair.")
-        owner_privkey_info = get_owner_privkey_info( wallet_keys=wallet_keys, config_path=config_path )
-        if owner_privkey_info is None:
-            raise Exception("No owner private key info")
-            return {'error': 'No usable private signing key found'}
-
-        # sanity check: must be a single private key 
-        if not is_singlesig( owner_privkey_info ):
-            raise Exception("Owner private key info must be a single key")
-            return {'error': 'No usable private signing key found'}
-
-        # sanity check: must match profile address
-        owner_pubkey = virtualchain.BitcoinPrivateKey(owner_privkey_info).public_key().to_hex()
-        compressed_addr, uncompressed_addr = get_pubkey_addresses( owner_pubkey )
-        if owner_address not in [compressed_addr, uncompressed_addr]:
-            raise Exception("%s not in [%s,%s]" % (owner_address, compressed_addr, uncompressed_addr))
-            return {'error': 'No usable public key'}
-
-        data_privkey = owner_privkey_info
+    data_privkey = get_data_privkey(user_zonefile, wallet_keys=wallet_keys, config_path=config_path)
+
+    if data_privkey is not None:
+        return {'status': True, 'privatekey': data_privkey}
+
+    # This is legacy code here.  The only time this should happen is
+    # when the user has a single owner key, and does not have a
+    # separate data key.
+    log.warn('No data private key set.  Falling back to owner keypair.')
+    owner_privkey_info = get_owner_privkey_info(wallet_keys=wallet_keys, config_path=config_path)
+    if owner_privkey_info is None:
+        # FIXME: We cannot both raise and return
```

Comment:
Thanks for catching that.  I added a `raise` to see if I could get this method to fail more loudly in this case and the case below.  I probably added it while testing, and just forgot to clean up the `return`.